src/main/java/com/google/cloud/tools/appengine/cloudsdk/serialization/CloudSdkVersionPreRelease.java

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright 2016 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.appengine.cloudsdk.serialization;
+
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Represents the pre-release field in a Cloud SDK Version. The pre-release field is comprised of
+ * one or more dot-separated segments, which are can be treated as either numeric-only, or
+ * alphanumeric, depending on their contents.
+ */
+class CloudSdkVersionPreRelease implements Comparable<CloudSdkVersionPreRelease> {
+
+  private List<PreReleaseSegment> segments;
+  private final String preRelease;
+
+  /**
+   * Constructs a new CloudSdkVersionPreRelease from a string representation.
```

Comment:
an example here of what a normal full semver string would be and which part is the pre-release could be helpful.